### ISSUE TITLE
Fix build for PyPy

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/tskit#859]
Conda-forge are building packages for pypy (see https://github.com/conda-forge/tskit-feedstock/pull/13) and it looks like things mostly work, except for the usage of PyStructSequence_SetItem:

```
_tskitmodule.c: In function 'TreeSequence_get_table_metadata_schemas':
  _tskitmodule.c:7172:9: warning: implicit declaration of function 'PyStructSequence_SetItem'; did you mean 'PyPySequence_SetItem'? [-Wimplicit-function-declaration]
           PyStructSequence_SetItem(value, j, schema);
           ^~~~~~~~~~~~~~~~~~~~~~~~
           PyPySequence_SetItem
```
Build details [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=213327&view=logs&j=9a864fd9-6c8f-52ca-79ce-2aa6dca1a1de&t=10fc5aa2-324e-5982-4c88-6b31fcab16b3&l=598)

This might be the tip of the iceberg, but if it's only this one issue, it might be worth working around it? I'm sure we could do without PyStructSequence if we needed to.

**Comment by benjeffery** [comment]
Ha, I remember you saying to avoid `PyStructSequence` if we didn't absolutely need it. I'll add that to the "list of things Jerome was right about". 
If we want to support PyPy it should be added to our CI - we shouldn't be finding out post-release here. I'll look into how to build against PyPy, self-assigning.

**Comment by jeromekelleher** [comment]
> I'll add that to the "list of things Jerome was right about".

Important to also keep the (longer) "list of things Jerome was wrong about"!

**Comment by benjeffery** [comment]
I've looked into this, even if I fix this issue there are others that prevent the test suite running on PyPy (h5py gives `ValueError: array.array size changed, may indicate binary incompatibility. Expected 72 from C header, got 24 from PyObject`). Trying to run tests that don't use hdf5 (test_tables) gives `Fatal error in cpyext, CPython compatibility layer: converting a PyTupleObject into a W_TupleObject, but found NULLs as items Aborted (core dumped)`

In summary I would like to support PyPy, but it isn't a quick fix and don't want to take it on now.

**Comment by jeromekelleher** [comment]
OK, good, thanks. Let's not worry about PyPy then.
